packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/modals/PromptStudentToLeaveQueueModal.tsx

```diff
@@ -1,21 +1,64 @@
-import { Button, message } from 'antd'
+import { Button, message, Tooltip } from 'antd'
 import Modal from 'antd/lib/modal/Modal'
-import { useState } from 'react'
-import { DoorOpen, PersonStanding } from 'lucide-react'
+import { useState, useEffect } from 'react'
+import { ArrowRightLeft, DoorOpen, PersonStanding } from 'lucide-react'
 import { API } from '@/app/api'
 import { getErrorMessage } from '@/app/utils/generalUtils'
 import { ClosedQuestionStatus } from '@koh/common'
+import { usePathname } from 'next/navigation'
+import Link from 'next/link'
 
 type PromptStudentToLeaveQueueModalProps = {
   qid: number
+  cid: number
   handleClose: () => Promise<void>
 }
 const PromptStudentToLeaveQueueModal: React.FC<
   PromptStudentToLeaveQueueModalProps
-> = ({ qid, handleClose }) => {
+> = ({ qid, cid, handleClose }) => {
   const [isStayLoading, setIsStayLoading] = useState(false)
   const [isLeaveLoading, setIsLeaveLoading] = useState(false)
+  const [isConvertLoading, setIsConvertLoading] = useState(false)
   const [modalJustOpened, setModalJustOpened] = useState(true)
+  const [queueQuestionId, setQueueQuestionId] = useState<number | null>(null)
+  const [canConvert, setCanConvert] = useState(false)
+  const [convertTooltip, setConvertTooltip] = useState(
+    'Demos cannot be converted into anytime questions',
+  )
+  const pathname = usePathname()
+
+  useEffect(() => {
+    const fetchQuestionId = async () => {
+      const questions = await API.questions.index(qid)
+      const myQuestions = questions.yourQuestions || []
+      let myQuestion = null
+
+      if (myQuestions.length === 1) {
+        myQuestion = myQuestions[0]
+        if (myQuestion.isTaskQuestion) {
+          setCanConvert(false)
+          setConvertTooltip('Demos cannot be converted into anytime questions')
+        } else {
+          setCanConvert(true)
+          setConvertTooltip('')
+        }
+      } else if (myQuestions.length > 1) {
+        myQuestion = myQuestions.find(
+          (q) => q.queueId === qid && q.isTaskQuestion === false,
+        )
```

Comment:
```suggestion
        myQuestion = myQuestions.find(
          (q) => q.isTaskQuestion === false,
        )
```
Questions don't have a .queueId attribute (it gives a linter error). Also that check is not necessary since the endpoint is only getting questions from this queue